Git js tree

hanumakanthvvn opened the pull request.

kanth committed:
new font moved to public dir

prasadsurase committed:
Added views for exercise pages.

kanth committed:
ui changes for static pages

code-R committed:
minor ui bug fixes for nav bar for logged in users

code-R committed:
use footer partial in hootcode layout

code-R committed:
require pry for test and deve environments

code-R committed:
override some bootstrap styles

code-R committed:
implementation git browsing with js tree

hanumakanthvvn committed:
modified git tree functionality

made changes for readability

added source classifier to identify language

test commit

Removed Github api gem

removed github-api gem

SaiVardhan committed:
updated css for jstree

code-R commented on Gemfile:
```diff
@@ -25,6 +25,8 @@ gem 'sidekiq'
 gem 'will_paginate'
 gem 'will_paginate-bootstrap'
 gem 'github_api'
```
can we remove this github_api gem ?

code-R commented on frontend/app/css/style.css:
```diff
@@ -0,0 +1,1031 @@
+/* jsTree default theme */
```
rename this style css file.

code-R commented on frontend/app/js/custom.js:
```diff
@@ -0,0 +1,18 @@
+$('#git_tree_content').on("select_node.jstree", function(e, data){
+  var git_node = data.node.data;
+  if(git_node.type === "file"){
+  $.ajax('blob/content', {
```
route should be restful (should be blobs/:sha) something..

code-R commented on frontend/app/js/custom.js:
```diff
@@ -0,0 +1,18 @@
+      $("#git_code").html(response.data);
+    },
+    error: function() {
+      alert("Error");
```
put more descriptive message if possible, send the error response from error, when there is exception in server side like git timeout or invalid sha something..

code-R commented on lib/app/routes/exercises.rb:
```diff
@@ -36,12 +37,19 @@ class Exercises < Core
+          data = {submission: submission,
+                  next_submission: next_submission,
+                  sharing: Sharing.new,
+                  solution: get_tree(submission)}
```
the logic should be moved to a class, rather than a method in routes

code-R commented on lib/app/routes/exercises.rb:
```diff
@@ -133,6 +141,52 @@ class Exercises < Core
         redirect "/"
       end
+
+      get '/submissions/blob/content' do
```
I think route should be /submissions/:key/blobs/:sha..

code-R commented on lib/app/routes/exercises.rb:
```diff
@@ -133,6 +141,52 @@ class Exercises < Core
+
+      get '/submissions/blob/content' do
+        content_type :json
+        submission = Submission.find_by_key(params[:key])
```
dont use dynamic finders (they are deprecated)

SaiVardhan committed:
modified git tree functionality

Conflicts:
	frontend/app/js/custom.js
	lib/app/public/js/app.js
	lib/app/public/js/custom.js
	lib/app/routes/exercises.rb
	lib/app/views/submissions/current_submission.erb
